doc/languages-frameworks/vim.section.md

```diff
@@ -154,6 +154,37 @@ in
 }
 ```
 
+### Specificities for some plugins
+#### Tree sitter
+
+By default `nvim-treesitter` encourages you to download, compile and install
+the required tree-sitter grammars at run time with `:TSInstall`. This works
+poorly on NixOS.  Instead, to install the `nvim-treesitter` plugins with a set
+of precompiled grammars, you can use `nvim-treesitter.withPlugins` function:
+
+```nix
+{ pkgs, ... }: {
+  environment.systemPackages = [
+    (pkgs.neovim.override {
+      configure = {
+        packages.myPlugins = with pkgs.vimPlugins; {
+          start = [
+            (nvim-treesitter.withPlugins (
+              plugins: with plugins; [
+                tree-sitter-nix
+                tree-sitter-python
+              ]
+            ))
+          ];
+        };
+      };
+    })
+  ];
+}
+```
+
+To enable all grammars packaged in nixpkgs, use `pkgs.vimPlugins.nvim-treesitter.withPlugins (plugins: pkgs.tree-sitter.allGrammars)`.
```

Comment:
```suggestion
To enable all grammars packaged in nixpkgs, use `(pkgs.vimPlugins.nvim-treesitter.withPlugins (plugins: pkgs.tree-sitter.allGrammars))`.
```

If you don't put this in brackets it will lead to

> error: cannot coerce a function to a string, at /etc/nixos/nixpkgs/pkgs/misc/vim-plugins/vim-utils.nix:189:44

Reply:
fixed